src/data/list/nodup.lean

```diff
@@ -28,6 +28,10 @@ namespace list
 @[simp] theorem nodup_cons {a : α} {l : list α} : nodup (a::l) ↔ a ∉ l ∧ nodup l :=
 by simp only [nodup, pairwise_cons, forall_mem_ne]
 
+protected lemma pairwise.nodup {l : list α} {r : α → α → Prop} [is_irrefl α r] (h : pairwise r l) :
+  nodup l :=
+h.imp $ λ a b hr heq, irrefl_of r a $ by rwa ← heq at hr
```

Comment:
I think the `irrefl_of` here should be a theorem like `r a b -> a != b`, say `ne_of_irrefl`

Reply:
[irrefl_of](https://leanprover-community.github.io/mathlib_docs/init/algebra/classes.html#irrefl_of) is in the core, so I can't rename it in this PR.

Reply:
irrefl_of doesn't need to change, I'm suggesting a new theorem with a different statement (I think `irrefl_of` says `not (r a a)`). I think the theorems about unbundled order classes goes in `order.basic`.